### ISSUE TITLE
Restyle expression-derived source with styler

### ISSUE THREAD

**Issue by jennybc** [tidyverse/reprex#94]
Expression input generally gets processed with `formatR::tidy_source()` due to the addition of this line in the `.R` source file:

```r
knitr::opts_chunk$set(tidy = TRUE, tidy.opts = list(indent = 2))
```

But this does not play well with the bang-bang syntactic sugar from tidyeval:

``` r
formatR::tidy_source(
  text = c(
    "nameshift <- c(SL = 'Sepal.Length')",
    "head(dplyr::rename(iris[,1:2], !!!nameshift), 3)"
  )
)
#> nameshift <- c(SL = "Sepal.Length")
#> head(dplyr::rename(iris[, 1:2], !(!(!nameshift))), 3)
```

[styler](https://github.com/krlmlr/styler#readme) is under development and its non-invasive quality works to our advantage:

``` r
styler::style_text(
  c(
    "nameshift <- c(SL = 'Sepal.Length')",
    "head(dplyr::rename(iris[,1:2], !!!nameshift), 3)"
  )
)
#> [1] "nameshift <- c(SL = \"Sepal.Length\")"            
#> [2] "head(dplyr::rename(iris[, 1:2], !!!nameshift), 3)"
```

Switch to that once it's on CRAN.

cc @krlmlr @lorenzwalthert

**Comment by lorenzwalthert** [comment]
Glad to see it works for you. We haven't payed particular attention to tidyeval / rlang features yet, so let us know if you find something missing / wrongly formatted. A CRAN submission is planned for late August.

**Comment by jennybc** [comment]
How firm is your timeline @lorenzwalthert? If it's pretty good, I could switch over now and have dev reprex depend on dev styler, which might allow a few of us to accumulate a bit of reprex + styler + tidyeval experience w/o even trying.

**Comment by krlmlr** [comment]
We should release to CRAN as part of the GSoC project, we have no reason to expect major delays. It would be awesome to give styler a bit of exposure before that, but I remember a few minor problems we wanted to fix before announcing a stable-ish version. @lorenzwalthert: Can you please comment?

**Comment by lorenzwalthert** [comment]
Yes, @krlmlr  is right. We fixed all remaining (minor) problems now except for token-dependent indention (i.e. indention that is not just the regular 2, 4, etc. spaces but depends on the length of other tokens). That means styler will format 
```r
a <- function(x,
              y) {
}
```
and the like as
```r
a <- function(x,
  y) {
}
```

However we should also be able to fix that (see [PR #104](https://github.com/krlmlr/styler/pull/106)) soon, so we will submit to CRAN by the end of August.

**Comment by jennybc** [comment]
So we can just start using it here with no fanfare, knowing we won't have a non-CRAN dependency for long. I will leave any active publicity to you two.

**Comment by lorenzwalthert** [comment]
Ok, great.

**Comment by krlmlr** [comment]
I think we could give styler a try now, it seems to work well and we're about to submit to GitHub. Should we use styler exclusively, or keep formatR as an option?

**Comment by jennybc** [comment]
I lean towards just switching over to styler.

I am trying to get googledrive submitted to CRAN and can do some maintenance here right after.

**Comment by jennybc** [comment]
Re-styling is no longer mandatory for expression input, so closing this.

Re-styling with styler will, instead, be made available for all input modes: #108.
